lib/assessment.js

```diff
@@ -39,7 +60,14 @@ var newArray2 = function newArray2(length, defaultsFunction) {
 // The function you write should return
 // the object passed in as an argument or the object created
 var addMethod = function addMethod(propertyName, method, obj) {
-
+  this.propertyName = method();     //DAN: confused on where prop and method go.
+  var method = function (obj){
+    if(!obj){
+      return new var newObj{};
+    } else{
+      return obj;
+    }
+  }
```

Comment:
Here's the idea, step by step.
1. Check to see if obj has been defined; set it to a blank object to start out with.
2. Add a new method to that obj with the name given.
3. Return the modified object.

Here's how that might look in code.
```javascript
if (!obj) { // if obj is undefined, then !obj is true
  obj = {};
}
obj[propertyName] = method;
return obj;
```
That first bit could also have been written as obj = obj || {}, since || will evaluate to the first truthy argument it gets.